src/Microsoft.AspNetCore.Hosting/Helper/Network.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Hosting.Helper
+{
+    public class Network
+    {
+        public static bool PortValidity(int port)
```

Comment:
We have test infrastructure that does something similar except it checks by binding to the port as if it were going to be the server. I think it even tries to bind to 0 which causes it to generate and return a random port.